### PR TITLE
Feat/15645086 join colony button

### PR DESCRIPTION
**Pull request by CzarekDryl**
https://tw.auroracreation.com/app/tasks/15645086

Added join colony button with animation.

### CODE REVIEW

File: src/components/common/Extensions/UserNavigation/UserNavigation.tsx [hunk 1]
```diff
@@ -17,10 +18,10 @@ const UserNavigation: FC<UserNavigationProps> = ({
   txButtons = null,
 }) => {
   const { wallet, user, connectWallet } = useAppContext();
-  const { formatMessage } = useIntl();
   const isTablet = useTablet();
   const { setOpenItemIndex, mobileMenuToggle } = useNavigationSidebarContext();
   const [, { toggleOff }] = mobileMenuToggle;
+  const isOnColonyRoute = window.location.pathname.includes('/colony/');
```
Comment (TomaszChmielPagepro) [inline src/components/common/Extensions/UserNavigation/UserNavigation.tsx:24]: Use `useMatch` instead

**Review by rdig — CHANGES_REQUESTED** [review]
- Icon is misaligned
Screenshot from 2023-11-13 13-36-50

- Join functionality is not triggering correctly, as it needs to fire off the mutation at the start, and run in parallel to the animation, not wait until the animation finishes and then fire it off

- Leaving a colony (unjoin) will bring back the button with the wrong state: text "Joined" (vs. "Join") and the icon being visible

**Review by krawczykkatarzyna — APPROVED** [review]

File: src/components/common/Extensions/UserNavigation/UserNavigation.tsx (hunk 1 above)
Comment (CzarekDryl) [reply src/components/common/Extensions/UserNavigation/UserNavigation.tsx]: `useMatch` used

**Comment by CzarekDryl** [comment]
@rdig All issue mentioned by you are fixed, please check it once more :)

**Review by KornelKwiatkowski — APPROVED** [review]

**Review by rdig — CHANGES_REQUESTED** [review]
It's now working (more or less) as expected but has a bit of weird behavior.

- Join a colony
- Leave a colony
- Join a colony again -- triggers an Apollo error

**Review by ArmandoGraterol — DISMISSED** [review]
LGTM 👍

**Comment by CzarekDryl** [comment]
@rdig  it was the same in the old UI, it's only in the case of a new user. I think it's better for someone from your team to check it.

image

**Comment by rdig** [comment]
>  I think it's better for someone from your team to check it.
?

 I did try it with new users, 3 of them actually.

**Comment by TomaszChmielPagepro** [comment]
@rdig Didn't just @CzarekDryl say that it only occurs for new users? 😅
This seems to be a problem with the old UI as well and the error suggests that it might be something wrong on the backend side

**Review by melyndav — CHANGES_REQUESTED** [review]
Hey @CzarekDryl thanks for your work on this!

A minor update:

1. The 16px padding is missing for both the wallet connected and not connected states:
image

**Review by melyndav — APPROVED** [review]
Great work @CzarekDryl, works as expected.

**Review by arrenv — APPROVED** [review]
Looks good, thank you @CzarekDryl

**Pull request merged**